rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -519,201 +359,163 @@ public String releaseNode(String sessionId, String url) throws RestServerExcepti
+                                       final List<String> attrs, final String timeRange) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
+            @Override
+            public InputStream apply(RestClient restClient) {
+                return restClient.getNodeMBeansHistory(sessionId, nodeJmxUrl, objectNames, attrs, timeRange);
+            }
+        });
+    }
 
-            switch (code) {
-                case 200:
-                    return ret;
-                default:
-                    throw new RestServerException(code, ret);
-            }
-        } catch (IOException e) {
-            throw new ServiceException("Failed to read server response", e);
-        } finally {
-            if (clientResponse != null) {
-                clientResponse.releaseConnection();
+    @Override
+    public String getStatHistory(final String sessionId, final String range) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
+            @Override
+            public InputStream apply(RestClient restClient) {
+                return restClient.getStatHistory(sessionId, range);
             }
-        }
+        });
     }
 
     @Override
-    public String getStatHistory(String sessionId, String range) throws RestServerException, ServiceException {
-        RestClient client = ProxyFactory.create(RestClient.class, RMConfig.get().getRestUrl(), executor);
-        ClientResponse<InputStream> clientResponse = null;
-        try {
-            clientResponse = client.getStatHistory(sessionId, range);
-            int code = clientResponse.getStatus();
-            String ret = convertToString(clientResponse.getEntity());
-
-            switch (code) {
-                case 200:
-                    return ret;
-                default:
-                    throw new RestServerException(code, ret);
+    public String executeNodeScript(final String sessionId, final String script, final String engine, final String nodeUrl) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
+            @Override
+            public InputStream apply(RestClient restClient) {
+                return restClient.executeNodeScript(sessionId, nodeUrl, script, engine);
             }
-        } catch (IOException e) {
-            throw new ServiceException("Failed to read server response", e);
-        } finally {
-            if (clientResponse != null) {
-                clientResponse.releaseConnection();
+        });
+    }
+
+    private RestClient getRestClientProxy() {
+        ResteasyClient client = new ResteasyClientBuilder().asyncExecutor(threadPool).build();
+        ResteasyWebTarget target = client.target(RMConfig.get().getRestUrl());
+
+        return target.proxy(RestClient.class);
```

Comment:
I tried to apply the same semantic as it was before my changes. I prefer to keep it as it is for now since reusing the same client could have many side effects. However you are right, that's something that could probably be improved.